spec-tests/src/main/java/org/candlepin/spec/bootstrap/data/util/CertificateUtil.java

```diff
@@ -0,0 +1,92 @@
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.spec.bootstrap.data.util;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.apache.commons.codec.binary.Base64;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.zip.DataFormatException;
+import java.util.zip.Inflater;
+
+public final class CertificateUtil {
+
+    private CertificateUtil() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Extracts the decoded and uncompressed bodies of the entitlement certificates.
+     *
+     * @param jsonPayload - the json that contains the encoded and compressed entitlement certificates
+     * @return the entitlement certificate bodies
+     * @throws DataFormatException if unable to decompress the body of the certificate
+     * @throws JsonMappingException if unable to parse the json
+     * @throws JsonProcessingException if unable to parse the json
+     */
+    public static List<JsonNode> extractEntitlementCertificatesFromPayload(Object jsonPayload)
+        throws DataFormatException, JsonMappingException, JsonProcessingException {
+        if (jsonPayload == null) {
+            return new ArrayList<>();
+        }
+
+        String json = String.valueOf(jsonPayload);
+        if (json == null || json.isEmpty() || json.isBlank()) {
+            return new ArrayList<>();
+        }
+
+        // Retrieves the certificate substrings from the json.
+        List<String> certificates = new ArrayList<>();
+        int certEndIndex = 0;
+        int certStartIndex = json.indexOf("cert=", certEndIndex);
+        while (certStartIndex != -1 && certEndIndex != -1) {
+            certEndIndex = json.indexOf(",", certStartIndex);
+            if (certEndIndex != -1) {
+                certificates.add(json.substring(certStartIndex, certEndIndex));
+                certStartIndex = json.indexOf("cert=", certEndIndex);
+            }
+        }
+
+        // Create json nodes for each decoded certificate
+        ObjectMapper objectMapper = new ObjectMapper();
+        List<JsonNode> jsonNodes = new ArrayList<>();
+        for (String certificate : certificates) {
+            // Retrieve the compressed data body
+            certificate = certificate.split("-----BEGIN ENTITLEMENT DATA-----\n")[1];
+            certificate = certificate.split("-----END ENTITLEMENT DATA-----")[0];
+            byte[] compressedBody = fromBase64(certificate.getBytes());
+
+            // Decompress the data
+            Inflater decompressor = new Inflater();
+            decompressor.setInput(compressedBody);
+            byte[] decompressedBody = new byte[1024];
```

Comment:
I'm not sure how we can know how much to allocate for the decompressed body here